system/lib/wasmfs/syscalls.cpp

```diff
@@ -513,4 +520,85 @@ __wasi_errno_t __wasi_fd_fdstat_get(__wasi_fd_t fd, __wasi_fdstat_t* stat) {
   }
   return __WASI_ERRNO_SUCCESS;
 }
+
+// This enum specifies whether rmdir or unlink is being performed.
+enum class UnlinkMode { Rmdir, Unlink };
+
+static long __unlink(char* path, UnlinkMode unlinkMode) {
+  auto pathParts = splitPath(path);
+
+  // Root (/) directory cannot be removed.
+  if (pathParts.size() == 1 && pathParts[0] == "/") {
+    return -EBUSY;
+  }
+
+  if (pathParts.empty() || pathParts.back() == ".") {
```

Comment:
I'm surprised that `.` is treated specially but `..` is not. Is that consistent with native implementations? It would probably be worth a comment mentioning that this is intentional because this is how the call is specified.

Reply:
Left a TODO comment since this should realistically be handled by the path parsing logic.